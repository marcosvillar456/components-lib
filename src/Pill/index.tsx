import './styles/__index.scss';
import Image from '../Image';
import classNames from 'classnames';
import type PillInterface from './types';

const Pill = ({
  icon,
  text,
  background,
  style,
}: PillInterface): JSX.Element => {
  const { label, color = '#FFF' } = text;

  const containerStyle = {
    ...style,
    '--background-pill': background,
    '--text-color-pill': color,
  };

  return (
    <div
      className={classNames('pill', {
        'pill--icon': !!icon,
      })}
      style={containerStyle}
    >
      {icon && (
        <div
          className={classNames('pill__image', {
            'pill__image--separator': !!icon.separator,
          })}
        >
          <Image alt={icon.alt} src={icon.src} />
        </div>
      )}
      <p className='pill__text'>{label}</p>
    </div>
  )
};

Pill.defaultProps = {
  text: {
    color: 'black',
  },
};

export default Pill;
