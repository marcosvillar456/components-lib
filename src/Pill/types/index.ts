import ImageInterface from 'Image/types';

interface iconInterface extends ImageInterface {
  separator: boolean;
}
interface PillInterface {
  icon?: iconInterface;
  text: {
    label: string;
    color?: string;
  };
  background: string;
  style?: React.CSSProperties & {
    '--background-pill': string;
    '--text-color-pill': string;
  };
}

export default PillInterface;
