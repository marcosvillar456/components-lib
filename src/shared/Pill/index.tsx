import "./styles/_index.scss";
import Image from "../Image";
import classNames from "classnames";
import type { PillInterface, stylesInlineInterface } from "./types";

const Pill = ({
  icon,
  text: { label, color = "#FFF" },
  background,
  card = false,
}: PillInterface): JSX.Element => {
  const containerStyle: stylesInlineInterface = {
    "--background-pill": background,
    "--text-color-pill": color,
  };

  return (
    <div
      className={classNames("pill", {
        "pill--icon": !!icon,
        "pill--card": card,
      })}
      style={containerStyle}
    >
      {icon && (
        <Image
          alt={icon.alt}
          src={icon.src}
          className={classNames("pill__image", {
            "pill__image--separator": !!icon.separator,
          })}
        />
      )}
      <p className="pill__text">{label}</p>
    </div>
  );
};

Pill.defaultProps = {
  text: {
    color: "black",
  },
};

export default Pill;
