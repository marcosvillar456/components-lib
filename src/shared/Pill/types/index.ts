import type I_Image from "../../Image/types";

interface I_StylesInline extends React.CSSProperties {
  "--background-pill": string;
  "--text-color-pill": string;
}

interface I_Icon extends I_Image {
  separator: boolean;
}

interface I_Pill {
  icon?: I_Icon;
  card?: boolean;
  text: {
    label: string;
    color?: string;
  };
  background: string;
}

export { I_Pill, I_StylesInline };
