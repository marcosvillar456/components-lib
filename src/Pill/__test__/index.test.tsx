import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Pill from "..";

describe("Pill component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("snapshot of component", () => {
    const props = {
      icon: {
        src: "path/to/image.png",
        alt: "icon",
        separator: true,
      },
      text: {
        label: "Hello world",
        color: "#fff",
      },
      background: "#000",
    };

    const { asFragment } = render(<Pill {...props} />);

    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class="pill pill--icon"
          style="--background-pill: #000; --text-color-pill: #fff;"
        >
          <div
            class="pill__image pill__image--separator"
          >
            <img
              alt="icon"
              height="24"
              src="path/to/image.png"
              width="24"
            />
          </div>
          <p
            class="pill__text"
          >
            Hello world
          </p>
        </div>
      </DocumentFragment>
    `);
  });

  it("renders the label text correctly (pill__text)", () => {
    const props = {
      text: {
        label: "Hello world",
      },
      background: "#000",
    };

    render(<Pill {...props} />);
    expect(screen.getByText(/Hello world/)).toBeInTheDocument();

  });

  it("renders the icon when passed as a prop", () => {
    const props = {
      icon: {
        src: "path/to/image.png",
        alt: "icon",
        separator: true,
      },
      text: {
        label: "Hello world",
        color: "#fff",
      },
      background: "#000",
    };

    render(<Pill {...props} />);
    expect(screen.getByAltText("icon")).toBeInTheDocument();
  });

  it("does not render the icon when not passed as a prop", () => {
    const props = {
      text: {
        label: "Hello world",
        color: "#fff",
      },
      background: "#000",
    };

    render(<Pill {...props} />);
    expect(screen.queryByAltText("icon")).toBeNull();
  });
});
