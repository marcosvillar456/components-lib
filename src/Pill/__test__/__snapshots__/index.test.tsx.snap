// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill component snapshot of component 1`] = `
<div
  className="pill pill--icon"
  style={
    Object {
      "--background-pill": "#000",
      "--text-color-pill": "#fff",
    }
  }
>
  <div
    className="pill__image pill__image--separator"
  >
    <img
      alt="icon"
      height={24}
      src="path/to/image.png"
      width={24}
    />
  </div>
  <p
    className="pill__text"
  >
    Hello world
  </p>
</div>
`;
